src/Microsoft.AspNet.Hosting/Internal/HostingLoggerExtensions.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Http.Features;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNet.Hosting.Internal
+{
+    internal static class HostingLoggerExtensions
+    {
+        public static IDisposable RequestScope(this ILogger logger, HttpContext httpContext)
+        {
+            return logger.BeginScopeImpl(new HostingRequestScope(httpContext));
+        }
+
+        public static void RequestStarting(this ILogger logger, HttpContext httpContext)
+        {
+            if (logger.IsEnabled(LogLevel.Information))
+            {
+                logger.Log(LogLevel.Information, 1, new HostingRequestStarting(httpContext), null, HostingRequestStarting.Callback);
+            }
+        }
+
+        public static void RequestFinished(this ILogger logger, HttpContext httpContext)
+        {
+            if (logger.IsEnabled(LogLevel.Information))
+            {
+                logger.Log(LogLevel.Information, 2, new HostingRequestFinished(httpContext), null, HostingRequestFinished.Callback);
+            }
+        }
+
+        private class HostingRequestScope : ILogValues
+        {
+            private readonly HttpContext _httpContext;
+
+            private FeatureReference<IHttpRequestIdentifierFeature> _requestIdentifierFeatureReference;
+            private IEnumerable<KeyValuePair<string, object>> _cachedGetValues;
+            private string _cachedToString;
+
+            public HostingRequestScope(HttpContext httpContext)
+            {
+                this._httpContext = httpContext;
+            }
+
+            public IHttpRequestIdentifierFeature RequestIdFeature =>
+                _requestIdentifierFeatureReference.Fetch(_httpContext.Features) ??
+                _requestIdentifierFeatureReference.Update(_httpContext.Features, new FastHttpRequestIdentifierFeature());
+
+            public override string ToString() => _cachedToString ?? Interlocked.CompareExchange(
+                ref _cachedToString,
+                $"RequestId:{RequestIdFeature.TraceIdentifier} RequestPath:{_httpContext.Request.Path}",
```

Comment:
For "why bother at all" it's that it's saving the manual step of taking a request id and scrolling back in the logs for the matching record that says what it connected to. It's a scenario specific value judgement.